crates/flox/src/commands/mod.rs

```diff
@@ -436,41 +441,67 @@ pub enum EnvironmentSelect {
     ),
-}
-
-impl Default for EnvironmentSelect {
-    fn default() -> Self {
-        EnvironmentSelect::Dir(PathBuf::from("./"))
-    }
+    #[default]
+    Unspecified,
 }
 
 impl EnvironmentSelect {
     pub fn to_concrete_environment(&self, flox: &Flox) -> Result<ConcreteEnvironment> {
-        let env = match self {
-            EnvironmentSelect::Dir(path) => {
-                let pointer = EnvironmentPointer::open(path)
-                    .with_context(|| format!("No environment found in {path:?}"))?;
-
-                match pointer {
-                    EnvironmentPointer::Path(path_pointer) => {
-                        debug!("detected concrete environment type: path");
-                        let dot_flox_path = path.join(DOT_FLOX);
-                        ConcreteEnvironment::Path(PathEnvironment::open(
-                            path_pointer,
-                            dot_flox_path,
-                            &flox.temp_dir,
-                        )?)
+        match self {
+            EnvironmentSelect::Dir(path) => Self::open_path(flox, path),
+            // If the user doesn't specify an environment, check if there's an
+            // already activated environment or an environment in the current
+            // directory.
+            // TODO: needs design - do we want to search up?
+            EnvironmentSelect::Unspecified => {
+                let maybe_activated = last_activated_environment();
+                let current_dir = env::current_dir().context("could not get current directory")?;
+                let maybe_current_pointer = EnvironmentPointer::open(&current_dir);
+                match (maybe_activated, maybe_current_pointer) {
+                    (Some(_activated), Ok(_current_dir_pointer)) => {
+                        todo!("needs design");
                     },
-                    EnvironmentPointer::Managed(managed_pointer) => {
-                        debug!("detected concrete environment type: managed");
-                        let env = ManagedEnvironment::open(flox, managed_pointer, path)?;
-                        ConcreteEnvironment::Managed(env)
+                    (Some(activated), Err(_)) => Self::open_path(flox, &activated),
+                    (None, Ok(current_dir_pointer)) => {
+                        Self::open_pointer(flox, &current_dir, current_dir_pointer)
+                    },
+                    (None, Err(e)) => {
+                        Err(e).context(format!("No environment found in {current_dir:?}"))?
                     },
                 }
             },
             EnvironmentSelect::Remote(_) => todo!(),
-        };
+        }
+    }
+
+    fn open_path(flox: &Flox, path: &PathBuf) -> Result<ConcreteEnvironment> {
+        let pointer = EnvironmentPointer::open(path)
+            .with_context(|| format!("No environment found in {path:?}"))?;
+
+        Self::open_pointer(flox, path, pointer)
+    }
 
+    fn open_pointer(
```

Comment:
```suggestion
    fn open_env_pointer(
```
Suggestion: "pointer" is a term that has specific meaning in systems languages like C/C++/Rust. If we're going to use the term "environment pointer" we need to make sure that it's unambiguous at the call site which kind of pointer we're talking about.